scipy/optimize/optimize.py

```diff
@@ -52,26 +52,29 @@
 
 
 class MemoizeJac(object):
-    """ Decorator that caches the value gradient of function each time it
+    """ Decorator that caches the value and gradient of function each time it
     is called. """
 
     def __init__(self, fun):
         self.fun = fun
         self.jac = None
+        self.value = None
         self.x = None
 
+    def compute_if_needed(self, x, *args):
```

Comment:
Can we have this as a private method, `def _compute_if_needed(`

Reply:
done.